File: src/main/java/no/fint/provider/springer/handlers/UpdateSkoleressursHandler.java
package no.fint.provider.springer.handlers;

import no.novari.fint.model.resource.utdanning.elev.SkoleressursResource;
import no.novari.fint.model.utdanning.elev.ElevActions;
import org.springframework.stereotype.Service;

import java.util.Collections;
import java.util.Set;

@Service
public class UpdateSkoleressursHandler extends UpdateHandler<SkoleressursResource> {

    public UpdateSkoleressursHandler() {
        super(SkoleressursResource.class);
    }

    @Override
    public Set<String> actions() {
        return Collections.singleton(ElevActions.UPDATE_SKOLERESSURS.name());
    }
}

File: src/main/kotlin/no/fint/provider/springer/handlers/UpdateSkoleressursHandler.kt
package no.fint.provider.springer.handlers

import com.fasterxml.jackson.databind.ObjectMapper
import no.novari.fint.model.resource.utdanning.elev.SkoleressursResource
import no.novari.fint.model.utdanning.elev.ElevActions
import org.springframework.data.mongodb.core.MongoTemplate
import org.springframework.stereotype.Service
import no.fint.provider.springer.storage.Wrapper

@Service
class UpdateSkoleressursHandler(
    mongoTemplate: MongoTemplate,
    wrapper: Wrapper,
    mapper: ObjectMapper
) : UpdateHandler<SkoleressursResource>(mongoTemplate, wrapper, mapper, SkoleressursResource::class.java) {
    override fun actions(): Set<String> = setOf(ElevActions.UPDATE_SKOLERESSURS.name)
}
